File: app/src/main/java/org/exallium/tradetracker/app/Screen.java
package org.exallium.tradetracker.app;


/**
 * These are all Fragments
 */
public enum Screen {

    NONE(0, R.string.screen_unknown),
    TRADES(1, R.string.screen_trades),
    CARD_SETS(2, R.string.screen_cards),
    PEOPLE(3, R.string.screen_people),
    STATS(4, R.string.screen_stats),
    CARDS(5, R.string.screen_cards),
    TRADE(6, R.string.screen_trade);

    private final int id;
    private final int nameRes;

    Screen(int id, int nameRes) {
        this.id = id;
        this.nameRes = nameRes;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return MainApplication.getInstance().getString(nameRes);
    }

    public static Screen getById(int id) {
        for (Screen screen : Screen.values()) {
            if (screen.id == id)
                return screen;
        }
        return NONE;
    }
}

File: app/src/main/kotlin/org/exallium/tradetracker/app/Screen.kt
package org.exallium.tradetracker.app

import android.content.Context

public enum class Screen(public val id: Int, private val nameRes: Int) {

    NONE : Screen(0, R.string.screen_unknown)
    TRADES : Screen(1, R.string.screen_trades)
    CARD_SETS : Screen(2, R.string.screen_cards)
    PEOPLE : Screen(3, R.string.screen_people)
    STATS : Screen(4, R.string.screen_stats)
    CARDS : Screen(5, R.string.screen_cards)
    TRADE : Screen(6, R.string.screen_trade)

    public fun getName(): String {
        return Context.getString(nameRes)
    }

    public fun getById(id: Int): Screen {
        for (screen in values()) {
            if (screen.id == id)
                return screen
        }
        return NONE
    }
}